### PR TITLE
Fix minikube flag parsing.

### PR DESCRIPTION
**Pull request by dlorenc**
We can't use flags in init like this. They're not parsed yet.

### CODE REVIEW

**Comment by dlorenc** [comment]
@minikube-bot test this please

File: cmd/minikube/cmd/root.go [hunk 1]
```diff
@@ -94,6 +94,20 @@ var RootCmd = &cobra.Command{
...
 		}
+
+		//TODO(r2d4): config should not reference API
+		clientType = configCmd.GetClientType()
+
+		logDir := pflag.Lookup("log_dir")
+		if !logDir.Changed {
+			logDir.Value.Set(constants.MakeMiniPath("logs"))
+		}
+		cobra.OnInitialize(initConfig)
```
Comment (r2d4) [inline cmd/minikube/cmd/root.go]: I'm not sure if this belongs in the prerun - I think this belongs in init()

Comment (dlorenc) [inline cmd/minikube/cmd/root.go]: Can you take another look?

Comment (r2d4) [inline cmd/minikube/cmd/root.go]: OnIntialization adds that function as an initializer which gets called here
https://github.com/spf13/cobra/blob/master/command.go#L593

That code is called before the PersistentPreRun, so it we won't actually run our initializer function.  

This was also originally in init() before a recent merge

**Pull request merged**